### PR TITLE
Add ability to list, retrieve and delete prescriptions 

### PR DESCRIPTION
**Pull request by toddkazakov**
- [BACK-948] Get prescription by id (clinician)
- [BACK-1290] Get prescription by id (patient)
- [BACK-950] Delete prescription by id (clinician)
- [BACK-951] List prescriptions (clinician)
- [BACK-952] List prescriptions (patient)

[BACK-948]: https://tidepool.atlassian.net/browse/BACK-948
[BACK-948]: https://tidepool.atlassian.net/browse/BACK-948
[BACK-948]: https://tidepool.atlassian.net/browse/BACK-948
[BACK-950]: https://tidepool.atlassian.net/browse/BACK-950
[BACK-951]: https://tidepool.atlassian.net/browse/BACK-951

[BACK-1290]: https://tidepool.atlassian.net/browse/BACK-1290
[BACK-952]: https://tidepool.atlassian.net/browse/BACK-952

### CODE REVIEW

File: prescription/test/prescription_accessor.go [hunk 1]
```diff
@@ -41,6 +84,45 @@ func (p *PrescriptionAccessor) CreatePrescription(ctx context.Context, userID st
 	return output.Prescription, output.Error
 }
 
+func (p *PrescriptionAccessor) ListPrescriptions(ctx context.Context, filter *prescription.Filter, pagination *page.Pagination) (prescription.Prescriptions, error) {
+	p.ListPrescriptionsInvocations++
```
Comment (derrickburns) [inline prescription/test/prescription_accessor.go:88]: Is this debugging?

If so, how about using prometheus counters.

File: prescription/store/mongo/prescription_session.go [hunk 2]
```diff
@@ -18,38 +22,195 @@ type PrescriptionSession struct {
...
+			selector["_id"] = bson.ObjectIdHex(filter.ID)
+		} else {
+			selector["_id"] = nil
+		}
+	}
+	if filter.State != "" {
+		selector["state"] = filter.State
+	}
+	if filter.CreatedTimeStart != nil {
+		selector["createdTime"] = bson.M{"$gt": filter.CreatedTimeStart}
```
Comment (derrickburns) [inline prescription/store/mongo/prescription_session.go:203]: This interval is open on both ends.  Is that intentional. Do you want to use a half-open interval?

File: prescription/store/mongo/prescription_session.go [hunk 3]
```diff
@@ -18,38 +22,195 @@ type PrescriptionSession struct {
...
+	}
+	if filter.CreatedTimeEnd != nil {
+		selector["createdTime"] = bson.M{"$lt": filter.CreatedTimeEnd}
+	}
+	if filter.ModifiedTimeStart != nil {
+		selector["latestRevision.attributes.modifiedTime"] = bson.M{"$gt": filter.ModifiedTimeStart}
```
Comment (derrickburns) [inline prescription/store/mongo/prescription_session.go:209]: Same question as above.

File: prescription/store/mongo/prescription_session.go (hunk 2 above)
Comment (toddkazakov) [reply prescription/store/mongo/prescription_session.go:203]: Not intentional - I'll get that fixed.

**Comment by toddkazakov** [comment]
Superseded by https://github.com/tidepool-org/platform/pull/493

**Pull request closed without merging**